config/initializers/zz_hidden_filters.rb

```diff
@@ -0,0 +1,17 @@
+# frozen-string-literal: true
+
+# this file has a name late in the alphabet because it needs to be loaded after
+# (at least load_ontologies.rb), since it uses Event, Material and Trainer objects
+# which may use these.
+
+# systemwide ignored facet on frontend
+IGNORED_FILTERS = %w[user].freeze
+
+# check the filter lists for correctness
+if TeSS::Config.solr_enabled
+  [Event, Material, Trainer].each do |klass|
+    unknown_facets = Set.new(TeSS::Config.solr_facets&.fetch(klass.table_name, [])) - klass.facet_keys
+
+    raise "unknown facets defined for #{klass.name}: #{unknown_facets}" if unknown_facets.any?
+  end
```

Comment:
I wonder if instead of iterating `[Event, Material, Trainer]` it should just check all the keys of `TeSS::Config.solr_facets` and try and get the class from the key. If it can't find the class, it indicates another type of error.

Reply:
yes, good point. added.